openjdk6-b22/jdk/src/share/classes/javax/swing/plaf/basic/BasicComboPopup.java

```diff
@@ -1232,11 +1232,9 @@ protected Rectangle computePopupBounds(int px,int py,int pw,int ph) {
             screenBounds = new Rectangle(p, toolkit.getScreenSize());
         }
 
+        if (ph > screenBounds.height && screenBounds.height > 500)
+            ph = screenBounds.height - 1;
```

Comment:
Form the comment, I understand we should fix the Y position but here we change the size (the height).

Reply:
I tested this code on different screen layouts: multiple screens with horizontal and vertical layout.  
Fix of Y position does not work here because the code above this line does not work correctly on multiple monitors.
In particular GraphicsConfiguration gc = comboBox.getGraphicsConfiguration() does not returns correct screen. 
At the same time correction of Y is not required here. Y is corrected at another place and we do not have problems with it.
Correction of  height is required. Without this fix Atoms popup can take more then one screen.